### PR TITLE
feat: Logger passed to Gbts and Orthogonal seeders

### PR DESCRIPTION
**Pull request by CarloVarni**
Passing logger also to the GBTS and Orthogonal seeders

## Summary by CodeRabbit

- **New Features**
	- Enhanced logging capabilities across various classes, including `GbtsTrackingFilter`, `SeedFinderGbts`, and `SeedFinderOrthogonal`, allowing for better tracking and debugging.
	- Introduced logger parameters in constructors for `SeedFinderGbts`, `SeedFinderOrthogonal`, and `SeedingOrthogonalAlgorithm`, improving configurability.

- **Bug Fixes**
	- Improved error handling and logging for negative covariance values in `GbtsTrackingFilter`.
	- Adjusted warning messages in `GbtsSeedingAlgorithm` for better clarity.

- **Refactor**
	- Updated member variable types to use smart pointers for better memory management in `SeedFinderGbts` and `SeedFinderOrthogonal`.
	- Streamlined initialization processes in `SeedingOrthogonalAlgorithm` and `GbtsSeedingAlgorithm` to enhance code clarity and efficiency.
	- Renamed logger instances for consistency across `SeedFilter` and `SeedFinder`.

### CODE REVIEW

File: Core/include/Acts/Seeding/SeedFinderGbts.ipp [hunk 1]
```diff
@@ -34,22 +34,25 @@ namespace Acts {
...
 SeedFinderGbts<external_spacepoint_t>::SeedFinderGbts(
     const SeedFinderGbtsConfig<external_spacepoint_t>& config,
-    const GbtsGeometry<external_spacepoint_t>& gbtsGeo)
-    : m_config(config) {
-  m_storage = new GbtsDataStorage(gbtsGeo);
-}
+    const GbtsGeometry<external_spacepoint_t>& gbtsGeo,
+    std::unique_ptr<const Acts::Logger> logger)
+    : m_config(config),
+      m_storage(new GbtsDataStorage(gbtsGeo)),
```
Comment (paulgessinger) [inline Core/include/Acts/Seeding/SeedFinderGbts.ipp:40]: If you're touching this anyway, maybe you can switch it to a unique pointer?

**Review by paulgessinger — DISMISSED** [review]

File: Core/include/Acts/Seeding/SeedFinderGbts.ipp [hunk 2]
```diff
@@ -34,22 +34,25 @@ namespace Acts {
...
 
 template <typename external_spacepoint_t>
 SeedFinderGbts<external_spacepoint_t>::~SeedFinderGbts() {
-  delete m_storage;
-
-  m_storage = nullptr;
+  if (m_storage) {
+    delete m_storage;
+    m_storage = nullptr;
+  }
```
[the remark this answers is not in the document]
Comment (paulgessinger) [reply Core/include/Acts/Seeding/SeedFinderGbts.ipp:48]: The AI agrees with me! 🥳

File: Core/include/Acts/Seeding/SeedFinderOrthogonal.hpp [hunk 3]
```diff
@@ -238,6 +245,16 @@ class SeedFinderOrthogonal {
...
+
+  /**
+   * @brief Get the logger.
+   */
+  const Logger &logger() const { return *m_logger; }
+
+  /**
+   * @brief The logger
+   */
+  std::unique_ptr<const Acts::Logger> m_logger{nullptr};
```
[the remark this answers is not in the document]
Comment (CarloVarni) [reply Core/include/Acts/Seeding/SeedFinderOrthogonal.hpp:257]: this is not true. We use `ACTS_VERBOSE` in the ipp file. e.g. `ACTS_VERBOSE("Creating seeds with Orthogonal strategy");`

**Review by paulgessinger — APPROVED** [review]
Ok for me!

**Pull request merged**